doc/api/module.md

```diff
@@ -693,8 +703,24 @@ register('./my-hooks.mjs', {
 });
 ```
 
+### In-thread Synchronous Customization Hooks
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active development
+
+`module.registerHooks()` is a lighter weight alternative to `module.register()`
+which takes hook functions directly and run them in the same thread where the
+modules are loaded. The hook functions are synchronous, which is necessary to
```

Comment:
I think for people who are already using `module.register()` and trying to figure out what the new API offers, this should be noted. We can remove it when `module.register()` actually gets removed, which would take a while because the usage out there already warrants a deprecation cycle, and in that case this would also serve as what all deprecation needs in the documentation - how to migrate to the alternative of the deprecated API.